compiler/src/main/kotlin/dev/zacsweers/metro/compiler/ir/BindingGraph.kt

```diff
@@ -43,29 +43,11 @@ internal class BindingGraph(private val metroContext: IrMetroContext) {
         )
         val existing = bindings.getValue(key)
         val duplicate = binding
-        val locations =
-          listOfNotNull(
-              existing.reportableLocation?.render(),
-              duplicate.reportableLocation?.render(),
-            )
-            .distinct()
-        when (locations.size) {
-          0 -> {
-            appendLine("├─ No binding source locations available, they may be contributed.")
-          }
-          1 -> {
-            appendLine("├─ Binding 1: ${locations[0]}")
-            appendLine("├─ Binding 2: Unknown source location, this may be contributed")
-          }
-          2 -> {
-            for ((i, location) in locations.withIndex()) {
-              appendLine("├─ Binding ${i + 1}: $location")
-            }
-          }
-        }
+        appendLine("├─ Binding 1: ${existing.getContributionLocationOrDiagnosticInfo()}")
+        appendLine("├─ Binding 2: ${duplicate.getContributionLocationOrDiagnosticInfo()}")
         appendBindingStack(bindingStack)
       }
-      val location = binding.reportableLocation ?: bindingStack.graph.location()
+      val location = bindingStack.graph.location()
```

Comment:
Not strictly necessary but I noticed that prioritizing the binding location here meant we were duplicating the info with the other binding info lines